sdk/ml/azure-ai-ml/azure/ai/ml/entities/_monitoring/signals.py

```diff
@@ -385,22 +381,23 @@ def __init__(
     def _from_rest_object(  # pylint: disable=too-many-return-statements
         cls, obj: RestMonitoringSignalBase
     ) -> Optional["MonitoringSignal"]:
+        res: Optional["MonitoringSignal"] = None
         if obj.signal_type == MonitoringSignalType.DATA_DRIFT:
-            return DataDriftSignal._from_rest_object(obj)
-        if obj.signal_type == MonitoringSignalType.DATA_QUALITY:
-            return DataQualitySignal._from_rest_object(obj)
-        if obj.signal_type == MonitoringSignalType.PREDICTION_DRIFT:
-            return PredictionDriftSignal._from_rest_object(obj)
-        if obj.signal_type == "ModelPerformanceSignalBase":
-            return ModelPerformanceSignal._from_rest_object(obj)
-        if obj.signal_type == MonitoringSignalType.FEATURE_ATTRIBUTION_DRIFT:
-            return FeatureAttributionDriftSignal._from_rest_object(obj)
-        if obj.signal_type == MonitoringSignalType.CUSTOM:
-            return CustomMonitoringSignal._from_rest_object(obj)
-        if obj.signal_type == MonitoringSignalType.GENERATION_SAFETY_QUALITY:
-            return GenerationSafetyQualitySignal._from_rest_object(obj)
-
-        return None
+            res = DataDriftSignal._from_rest_object(obj)
+        elif obj.signal_type == MonitoringSignalType.DATA_QUALITY:
+            res = DataQualitySignal._from_rest_object(obj)
+        elif obj.signal_type == MonitoringSignalType.PREDICTION_DRIFT:
+            res = PredictionDriftSignal._from_rest_object(obj)
+        elif obj.signal_type == "ModelPerformanceSignalBase":
+            res = ModelPerformanceSignal._from_rest_object(obj)
+        elif obj.signal_type == MonitoringSignalType.FEATURE_ATTRIBUTION_DRIFT:
```

Comment:
Why are these being cast as MonitoringSignal instead of the specific type they are? (e.g. GenerationSafetyQualitySignal)

They should be left as whatever they are (no casting) and the return type annotation for this method should be the Union of all the possible types.